File: packages/sdk/src/resources/wallet.ts
import type { HttpClient } from '../http-client.js';
import type { Balance } from '../types.js';

export class WalletResource {
  constructor(private readonly http: HttpClient) {}

  /** GET /balance */
  getBalance(): Promise<Balance> {
    return this.http.request<Balance>('GET', '/balance');
  }
}

File: packages/sdk-python/src/smsgecko/resources/wallet.py
from __future__ import annotations

from .._http import HttpClient
from ..types import Balance


class WalletResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def get_balance(self) -> Balance:
        """GET /balance"""
        return Balance._from_dict(self._http.request("GET", "/balance"))
